packages/pcd/gpc-pcd/src/GPCPCDPackage.ts

```diff
@@ -62,13 +65,42 @@ async function checkProofArgs(args: GPCPCDArgs): Promise<{
   }
   const proofConfig = deserializeGPCProofConfig(args.proofConfig.value);
 
-  if (!args.pod.value) {
+  // Filter out PODPCD records.
+  const podRecord = Object.fromEntries(
+    Object.entries(args).filter(([argName, _arg]) =>
+      argName.startsWith(`${PODPCD_ARG_PREFIX}_`)
+    )
+  ) as PODPCDRecordArg;
+
+  if (Object.keys(podRecord).length === 0) {
     throw new Error("No PODPCD value provided");
   }
-  const podPCD = await PODPCDPackage.deserialize(args.pod.value.pcd);
-  if (!isPODPCD(podPCD)) {
-    throw new Error("Wrong PCD type provided for PODPCD");
-  }
+
+  const pods = Object.fromEntries(
+    await Promise.all(
+      Object.entries(podRecord).map(async ([prefixedPODName, podPCDArg]) => {
+        const podNameMatch = prefixedPODName.match(PODPCD_ARG_REGEXP);
```

Comment:
I think it would be better to check the prefix separately, then check the PODName using POD_NAME_REGEX (or checkPODName).  Otherwise PODPCD_ARG_REGEX becomes something we could easily miss changing if/when POD_NAME_REGEX changes.